09-schema/00-overview.md

```diff
@@ -5,29 +5,85 @@ permalink: /docs/schema/overview
+We refer to data instances as things - they can be thought of as instances of types defined in the schema.
+
+### Types
+Types constitute the core of your schema. They provide the necessary vocabulary to talk about our domain. They come in three flavours: [Entities](/docs/schema/concepts#entity), [Relationships](/docs/schema/concepts#relationship), and [Attributes](/docs/schema/concepts#attribute):
+
+__Entities__ are the main actors in our domain. These are usually the type of things we want to know about. Entity types provide means of classifying the objects in our domain.
 
-Lastly, we have the [**Graql Rules**](/docs/schema/rules). Rules watch for schema-driven patterns in the data and infer queryable conclusions. Rules are one way to perform automated reasoning in a Grakn knowledge graph.
+__Relationships__ are things that connect other concepts. Each relationship can connect a number of things, as specified in your schema. The character of participation in a relationship is characterised by roles that can be played in that relationship. Each relationship is required to
+have at least one role.
+
+__Attributes__ are used to characterise concepts with small pieces of data to other (think of numbers, strings, dates etc.). They allow to attach values of specified datatype to our instances.
+
+Apart from serving as a mean of classification, types also define behaviours of their instances. Consequently, types can define the following behaviours:
+
+__has [attribute type]__ - the ability to have an attribute of specified type attached to an instance
+
+__plays [role]__ - the ability to participate in relationships that allow for that role to be played
+
+__relates [role]__ (only relationships) - the ability of other instances to play this role in instances of this relationship type
+
+### Type Hierarchies
+Besides the modularity that the concept types provide, we are free to form subtype relations between concept types. For a given child concept type subtyping a parent concept type, the child concept type inherits the attributes owned and roles played by the parent type.
+The mechanism is analogous of subclassing known in Object Oriented Programming. Each concept type can have only a single parent type - multiple inheritance is not supported. 
+Subtyping not only allows us to mirror the true nature of a dataset as perceived in the real world, but also enables automated reasoning.
+
+### Roles
+_Roles_ specify the nature of the connection between instances. They are not types themselves: you cannot have a thing which is an instance of a role, but you will be able to have things playing a role in a specific relationship. In your schema, we will need to specify what role relates to each relationship type and who can play this role. Thanks to roles, you will be able to guarantee the logical integrity of your data, avoiding to have a marriage between a cat and a building, for example, unless you specifically allow such a thing in the schema.
```

Comment:
_avoiding ~~to have~~ **having** a_ 
overall a bit complex, and I think we should make clear which are types in a schema using ` `, maybe this:
_integrity of your data, for example, disallowing a `marriage` between a `person` and a `building`, for example_

IMO, roles are analogous to methods in OOP, if we agree then we could include this too